sdc/datatypes/sdc_function_templates.py

```diff
@@ -0,0 +1,268 @@
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+"""
+
+| This file contains function templates used by the auto-generation script during build
+
+"""
+
+# below imports are copied into the auto-generated source file as-is
+# for the auto-generation script to work ensure they are not mixed up with code
+import numba
+import numpy
+import operator
+import pandas
+
+from numba.errors import TypingError
+from numba.extending import overload, overload_method, overload_attribute
+from numba import types
+
+import sdc
+from sdc.datatypes.common_functions import TypeChecker
+from sdc.datatypes.common_functions import (check_index_is_numeric, find_common_dtype_from_numpy_dtypes,
+                                            sdc_join_series_indexes, sdc_check_indexes_equal, check_types_comparable)
+from sdc.hiframes.pd_series_ext import SeriesType
+from sdc.str_arr_ext import (string_array_type, num_total_chars, str_arr_is_na)
+
+
+def sdc_pandas_series_operator_binop(self, other):
+    """
+    Pandas Series operator :attr:`pandas.Series.binop` implementation
+
+    Note: Currently implemented for numeric Series only.
+        Differs from Pandas in returning Series with fixed dtype :obj:`float64`
+
+    .. only:: developer
+
+    **Test**: python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_op1*
+              python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_op2*
+              python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_operator_binop*
+
+    Parameters
+    ----------
+    series: :obj:`pandas.Series`
+        Input series
+    other: :obj:`pandas.Series` or :obj:`scalar`
+        Series or scalar value to be used as a second argument of binary operation
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+        The result of the operation
+    """
+
+    _func_name = 'Operator binop().'
+
+    ty_checker = TypeChecker('Operator binop().')
+    ty_checker.check(self, SeriesType)
```

Comment:
Please, replace this check with:

```suggestion
    if not isinstance(self, SeriesType):
        return None
```

Reply:
Done

Reply:
Fixed once again)